tests/test_eval_detection.py

```diff
@@ -0,0 +1,91 @@
+    )
+
+
+@pytest.fixture
+def metrics(evaluator: DetectionEvaluator) -> DataFrame:
+    return evaluator.evaluate()
+
+
+def test_center_similarity() -> None:
+    """Test that the Center similarity function works."""
+    olr1 = np.array([ObjectLabelRecord(np.array([0, 0, 0, 0]), np.array([0, 0, 0]), 5.0, 5.0, 5.0, 0)])
+    olr2 = np.array([ObjectLabelRecord(np.array([0, 0, 0, 0]), np.array([3, 4, 0]), 5.0, 5.0, 5.0, 0)])
+    assert compute_affinity_matrix(olr1, olr2, SimFnType.CENTER) == -5
+
+
+def test_translation_distance() -> None:
+    """Test that the translation distance function works."""
+    df1 = DataFrame([{"translation": [0.0, 0.0, 0.0]}])
+    df2 = DataFrame([{"translation": [5.0, 5.0, 5.0]}])
+    assert dist_fn(df1, df2, DistFnType.TRANSLATION) == 75 ** (1 / 2)
+
+
+def test_scale_distance() -> None:
+    """Test that the scale distance function works."""
+    df1 = DataFrame([{"width": 5, "height": 5, "length": 5}])
+    df2 = DataFrame([{"width": 10, "height": 10, "length": 10}])
+    assert (dist_fn(df1, df2, DistFnType.SCALE) == 1 - 0.125).all()
+
+
+def test_orientation_distance() -> None:
+    """Test that the orientation distance function works."""
+    # check all of the 45 degree angles
+    vecs_45_apart = [angle * np.array([0, 0, 1]) for angle in np.arange(0, 2 * np.pi, np.pi / 4)]
+    for i in range(len(vecs_45_apart) - 1):
+        df1 = DataFrame([{"quaternion": quat_scipy2argo_vectorized(R.from_rotvec(vecs_45_apart[i]).as_quat())}])
+        df2 = DataFrame([{"quaternion": quat_scipy2argo_vectorized(R.from_rotvec(vecs_45_apart[i + 1]).as_quat())}])
+        assert np.isclose(dist_fn(df1, df2, DistFnType.ORIENTATION), np.pi / 4)
+        assert np.isclose(dist_fn(df2, df1, DistFnType.ORIENTATION), np.pi / 4)
+    # check all of the 90 degree angles
+    vecs_90_apart = [angle * np.array([0, 0, 1]) for angle in np.arange(0, 2 * np.pi, np.pi / 2)]
+    for i in range(len(vecs_90_apart) - 1):
+        df1 = DataFrame([{"quaternion": quat_scipy2argo_vectorized(R.from_rotvec(vecs_90_apart[i]).as_quat())}])
+        df2 = DataFrame([{"quaternion": quat_scipy2argo_vectorized(R.from_rotvec(vecs_90_apart[i + 1]).as_quat())}])
+        assert np.isclose(dist_fn(df1, df2, DistFnType.ORIENTATION), np.pi / 2)
+        assert np.isclose(dist_fn(df2, df1, DistFnType.ORIENTATION), np.pi / 2)
+
+
+def test_ap(metrics: DataFrame) -> None:
+    """Test that AP is 1 for the self-compared results."""
+    assert metrics.AP.loc["Average Metrics"] == 1
+
+
+def test_translation_error(metrics: DataFrame) -> None:
+    """Test that ATE is 0 for the self-compared results."""
+    assert metrics.ATE.loc["Average Metrics"] == 0
+
+
+def test_scale_error(metrics: DataFrame) -> None:
+    """Test that ASE is 0 for the self-compared results."""
+    assert metrics.ASE.loc["Average Metrics"] == 0
+
+
+def test_orientation_error(metrics: DataFrame) -> None:
```

Comment:
could we add a unit test with non-zero orientation error (and for each metric?)

Reply:
@seanremy 😄